hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -875,6 +916,86 @@ ChangesetInfoPtr XmlChangeset::splitChangeset(ChangesetInfoPtr changeset)
+            split->add(element_type, (ChangesetType)current_type, way->id());
+            changeset->remove(element_type, (ChangesetType)current_type, way->id());
+            return split;
+          }
+        }
+      }
+      else if (element_type == ElementType::Relation)
+      {
+        for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
+        {
+          if (changeset->contains(element_type, (ChangesetType)current_type, element_id))
+          {
+            XmlRelation* relation = dynamic_cast<XmlRelation*>(_allRelations[element_id].get());
+            //  Add the relation to the split and remove from the changeset
+            split->add(element_type, (ChangesetType)current_type, relation->id());
+            changeset->remove(element_type, (ChangesetType)current_type, relation->id());
+            return split;
+          }
+        }
+      }
+    }
+    //  See if the hint is something like: Relation with id  cannot be saved due to Relation with id 1707699
+    if (matchesRelationFailure(splitHint, element_id, member_id, member_type))
+    {
+      if (element_id != 0)
+      {
+        //  If there is a relation id, move just that relation to the split
+        for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
+        {
+          if (changeset->contains(member_type, (ChangesetType)current_type, element_id))
+          {
+            XmlRelation* relation = dynamic_cast<XmlRelation*>(_allRelations[element_id].get());
+            //  Add the relation to the split and remove from the changeset
+            split->add(ElementType::Relation, (ChangesetType)current_type, relation->id());
+            changeset->remove(ElementType::Relation, (ChangesetType)current_type, relation->id());
+            return split;
+          }
+        }
+      }
+      else
+      {
+        //  If no relation id is found, move all relations that contain the id/type combination to the split
+        for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
+        {
+          for (XmlElementMap::iterator it = _relations[current_type].begin(); it != _relations[current_type].end(); ++it)
+          {
+            XmlRelation* relation = dynamic_cast<XmlRelation*>(it->second.get());
+            //  Make sure that the changeset contains this relation and this relation contains the problematic element
+            if (relation->hasMember(member_type, member_id) &&
+                changeset->contains(ElementType::Relation, (ChangesetType)current_type, relation->id()))
+            {
+              //  Add the relation to the split and remove from the changeset
+              split->add(ElementType::Relation, (ChangesetType)current_type, relation->id());
+              changeset->remove(ElementType::Relation, (ChangesetType)current_type, relation->id());
+              return split;
```

Comment:
In this case, the relation ID is unknown (because it is a negative placeholder ID) so we iterate the relations looking for any that have a member that matches the type/ID of the unknown element causing the failure.  Grab the first relation found and split that out.  Continuing to add others causes other parts of the code to fail, so grabbing one at a time will guarantee success.